pipeline/frontend/yaml/types/base/bool.go

```diff
@@ -42,7 +42,16 @@ func (b *BoolTrue) UnmarshalYAML(value *yaml.Node) error {
 	return nil
 }
 
+// MarshalYAML implements custom Yaml marshaling.
+func (b BoolTrue) MarshalYAML() (any, error) {
+	return b.Bool(), nil
+}
+
 // Bool returns the bool value.
 func (b BoolTrue) Bool() bool {
 	return !b.value
 }
+
+func ToBoolTrue(v bool) BoolTrue {
+	return BoolTrue{value: !v}
```

Comment:
I don't think this is correct. `BoolTrue` just means the default is true. So `var: false` still sets `false`.

Reply:
see Bool() at line 52(new file) and line 37